backend/authentication/providers/providers.go

```diff
@@ -0,0 +1,93 @@
+package providers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/sensu/sensu-go/api/core/v2"
+)
+
+// Provider represents an abstracted authentication provider
+type Provider interface {
+	Authenticate(ctx context.Context, username, password string) (*v2.Claims, error)
+	GetName() string
+	Refresh(ctx context.Context, providerClaims v2.ProviderClaims) (*v2.Claims, error)
+	Type() string
+}
+
+// ID returns a unique identifier for a given provider
+func ID(p Provider) string {
```

Comment:
Would it make more sense to define this function in `backend/authentication/providers/basic/basic.go` as `func (p *Provider) ID() string`?